src/Workspaces/Core/Portable/Utilities/NullableHelpers/NullableExtensions.cs

```diff
@@ -78,5 +77,44 @@ public static NullableAnnotation GetNullability(this ITypeSymbol typeSymbol)
                 return typeSymbol;
             }
         }
+
+        public static ITypeSymbol GetConvertedTypeWithFlowNullability(this TypeInfo typeInfo)
+            => typeInfo.ConvertedType?.WithNullability(typeInfo.Nullability.FlowState);
+
+        public static ITypeSymbol GetConvertedTypeWithAnnotatedNullability(this TypeInfo typeInfo)
+            => typeInfo.ConvertedType?.WithNullability(typeInfo.Nullability.Annotation);
+
+        public static ITypeSymbol GetTypeWithFlowNullability(this TypeInfo typeInfo)
+            => typeInfo.Type?.WithNullability(typeInfo.Nullability.FlowState);
+
+        public static ITypeSymbol GetTypeWithAnnotatedNullability(this TypeInfo typeInfo)
+            => typeInfo.Type?.WithNullability(typeInfo.Nullability.Annotation);
+
+        public static ITypeSymbol GetTypeWithAnnotatedNullability(this IFieldSymbol fieldSymbol)
+            => fieldSymbol.Type.WithNullability(fieldSymbol.NullableAnnotation);
+
+        public static ITypeSymbol GetTypeWithFlowNullability(this IFieldSymbol fieldSymbol)
```

Comment:
Why the flow nullability overloads? The flow/annotated distinction only applies in the TypeInfo cases, so we don't need both sets of methods here.

Reply:
(this might have just been you being overly eager with copy/paste. :smile:)

Reply:
Yup, just copy paste bananza here.